packages/application-generic/src/usecases/get-subscriber-global-preference/get-subscriber-global-preference.usecase.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import {
+  PreferenceLevelEnum,
+  SubscriberEntity,
+  SubscriberPreferenceRepository,
+  SubscriberRepository,
+} from '@novu/dal';
+
+import { GetSubscriberGlobalPreferenceCommand } from './get-subscriber-global-preference.command';
+import { buildSubscriberKey, CachedEntity } from '../../services/cache';
+import { ApiException } from '../../utils/exceptions';
+
+@Injectable()
+export class GetSubscriberGlobalPreference {
+  constructor(
+    private subscriberPreferenceRepository: SubscriberPreferenceRepository,
+    private subscriberRepository: SubscriberRepository
+  ) {}
+
+  async execute(command: GetSubscriberGlobalPreferenceCommand) {
+    const subscriber =
+      command.subscriber ??
+      (await this.fetchSubscriber({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }));
+
+    if (!subscriber) {
+      throw new ApiException(`Subscriber ${command.subscriberId} not found`);
+    }
+
+    const subscriberPreference =
+      await this.subscriberPreferenceRepository.findOne({
+        _environmentId: command.environmentId,
+        _subscriberId: subscriber._id,
+        level: PreferenceLevelEnum.GLOBAL,
+      });
+
+    const subscriberChannelPreference = subscriberPreference?.channels;
+
+    return {
+      preference: {
+        enabled: subscriberPreference?.enabled ?? true,
+        channels: subscriberChannelPreference ?? {},
```

Comment:
I wonder if we could add here a default value like { email : true, ...} this way all of this use-case clients could consume this default and don't need to generate it themselves. (for example the addition logic like we have in SendMessage.stepPreferred)